### PR TITLE
refactor: prompts for UTG

### PR DESCRIPTION
**Pull request by khareyash05**
Update Prompts by giving more in depth clarity for functions
Give out sample test fnction format for each language while tackling corner cases 

Handle when refactored code is empty

Will need to get `app_modify` branch merged to incorporate full changes

### CODE REVIEW

File: pkg/service/utgen/assets/test_generation.toml [hunk 1]
```diff
@@ -128,30 +212,30 @@ new_tests:
...
-    "import { assert } from 'assert';"
-    "import { myFunction } from 'my_module';"
+    import { assert } from 'assert';
+    import { myFunction } from 'my_module';
     {%- endif %}
   library_installation_code: |
     {%- if language == "python" %}
-    pip install pytest
+    pip install pytest pytest-asyncio
     {%- elif language == "java" %}
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:239]: this was a sampler any reason of adding asyncio, if we remove it would it cause any issue?

File: pkg/service/utgen/assets/test_generation.toml [hunk 2]
```diff
@@ -180,6 +267,9 @@ Use block scalar('|') to format each YAML output.
 {% endif %}
 
 
+In Python, since this can be an issue Unclosed client session client_session: <aiohttp.client.ClientSession object at 0x102fed130>, please ensure using pytest-aiohttp to solve this issue
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:270]: I don't think this is correct

File: pkg/service/utgen/assets/test_generation.toml (hunk 1 above)
Comment (khareyash05) [reply pkg/service/utgen/assets/test_generation.toml:239]: Ya that was just for the decorator, I handled it below, will remove it

File: pkg/service/utgen/assets/test_generation.toml (hunk 2 above)
Comment (khareyash05) [reply pkg/service/utgen/assets/test_generation.toml:271]: Tests are failing due to unclosed session, I isolated it

Comment (shivamsouravjha) [reply pkg/service/utgen/assets/test_generation.toml:271]: we can do a better thing, please check if we send this as error, print the prompt probably it should have had the error (this seems bit focused with address)

File: pkg/service/utgen/assets/test_generation.toml [hunk 3]
```diff
@@ -128,26 +212,27 @@ new_tests:
...
     {%- if language == "python" %}
-    "import pytest"
-    "from my_module import my_function"
+    import pytest
     {%- elif language == "java" %}
-    "import org.junit.jupiter.api.Test;"
-    "import my.package.MyFunction;"
+    import org.junit.jupiter.api.Test;
+    import my.package.MyFunction;
     {%- elif language == "golang" %}
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:227]: any reason for chaning these ?

File: pkg/service/utgen/assets/test_generation.toml [hunk 4]
```diff
@@ -128,14 +212,15 @@ new_tests:
 {%- endif %}
   test_code: |
 {%- if language in ["python"] %}
+    import pytest
+    @pytest.mark.asyncio
     def ...
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:217]: why are we making this change?

File: pkg/service/utgen/assets/test_generation.toml [hunk 5]
```diff
@@ -128,14 +212,15 @@ new_tests:
...
 {%- else %}
     ...
 {%- endif %}
   new_imports_code: |
     {%- if language == "python" %}
     "import pytest"
-    "from my_module import my_function"
     {%- elif language == "java" %}
     "import org.junit.jupiter.api.Test;"
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:225]: any reason for removing this? i put it there so that it can get idea of both type of import formats

File: pkg/service/utgen/assets/test_generation.toml [hunk 6]
```diff
@@ -168,9 +253,79 @@ new_tests:
...
   test_tags: happy path
-    ...
+
+refactored_source_code: |
+  # Here is the modified source code that retains original functionality but improves testability.
+  ...
 ```
 
+additions:
+  additional_instructions_text: |
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:263]: we already have something in the name of additional_instructions_text , might create confusion , also instead of sending all languages for all ,let's find out the language and send text accordingly

File: pkg/service/utgen/assets/test_generation.toml (hunk 5 above)
Comment (khareyash05) [reply pkg/service/utgen/assets/test_generation.toml:225]: Thanks, wasnt able to find one issue just now, i guess this will solve it

File: pkg/service/utgen/assets/test_generation.toml [hunk 7]
```diff
@@ -180,6 +335,11 @@ Use block scalar('|') to format each YAML output.
 {% endif %}
 
 
+In Python, Since  async def functions are not natively supported for testing and can be skipped, please add the decorator @pytest.mark.asyncio to the test functions that are async def functions, also if you are using the decorator, please import pytest and pytest-asyncio as well.
+In Python, use absolute imports instead of relative imports to avoid issues with module resolution.
+
+In Golang, please import testing library and use the testing.T object to run the tests.
+
```
Comment (shivamsouravjha) [inline pkg/service/utgen/assets/test_generation.toml:342]: any reason of keep this here in this format?

hopefully we don't give it a lot of texts, lets try to keep it crisp and concise, as a lot of instruction might not be the actual help

Comment (khareyash05) [reply pkg/service/utgen/assets/test_generation.toml:342]: no not specific reason, could you please tell me where can i keep this, wasnt able to figure out

File: pkg/service/utgen/assets/test_generation.toml (hunk 4 above)
Comment (khareyash05) [reply pkg/service/utgen/assets/test_generation.toml:217]: Pytest wasnt. being added, i guess the comment below will solve the problem of async def functions

**Comment by khareyash05** [comment]
Closing this PR as it will be merged with https://github.com/keploy/keploy/pull/2411
